### PR TITLE
WW-347b | tracking create page flow via create button on non existing page

### PR DESCRIPTION
**Pull request by rybmat**
https://github.com/Wikia/app/pull/11587
https://wikia-inc.atlassian.net/browse/WW-347

//cc @Wikia/west-wing

### CODE REVIEW

File: extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.init.js [hunk 1]
```diff
@@ -228,9 +227,8 @@
 			ve.track( 'mwedit.init', { type: 'page', mechanism: 'click' } );
 
 			if ( history.pushState && uri.query.veaction !== 'edit' ) {
-				if (!pageExists) {
-					veEditUri.extend( { flow: 'create-page-create-button' } );
-				}
+				mw.hook('ve.afterVEInit').fire(veEditUri);
+
 				// Replace the current state with one that is tagged as ours, to prevent the
 				// back button from breaking when used to exit VE. FIXME: there should be a better
```
Comment (WikiaTech) [inline extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.init.js:233]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1134)

File: extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.init.js [hunk 2]
```diff
@@ -125,7 +125,6 @@
 	conf = mw.config.get( 'wgVisualEditorConfig' );
 	tabMessages = conf.tabMessages;
```
Comment (WikiaTech) [inline extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.init.js:126]: MAJOR Remove this useless assignment to local variable "tabMessages" [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1854)

File: extensions/wikia/FlowTracking/FlowTracking.hooks.php [hunk 3]
```diff
@@ -86,6 +90,31 @@ public static function getParamsFromUrlQuery( $url ) {
 		return $queryParams;
 	}
 
+	/**
+	 * @param PageHeaderController $header
+	 * @param $actions
+	 * @return bool
+	 */
+	public static function onBeforePrepareActionButtons( $header, &$actions ) {
```
Comment (WikiaTech) [inline extensions/wikia/FlowTracking/FlowTracking.hooks.php:98]: MAJOR Remove the unused function parameter "$header". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 37 issues
- MAJOR 25 major
- MINOR 6 minor
- INFO 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [ve.init.wikia.ViewPageTarget.init.js#L57](https://github.com/Wikia/app/blob/a723af64acd5fd0415cffe6343e597b50e31dc2c/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.init.js#L57): Make this function anonymous by removing its name: 'function() {...}'. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
2. MAJOR [ve.init.wikia.ViewPageTarget.init.js#L143](https://github.com/Wikia/app/blob/a723af64acd5fd0415cffe6343e597b50e31dc2c/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.init.js#L143): Reduce the number of conditional operators (12) used in the expression (maximum allowed 3). [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1067)
3. MAJOR [ve.init.wikia.ViewPageTarget.init.js#L217](https://github.com/Wikia/app/blob/a723af64acd5fd0415cffe6343e597b50e31dc2c/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.init.js#L217): Function has a complexity of 11 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
4. MAJOR [ve.init.wikia.ViewPageTarget.init.js#L222](https://github.com/Wikia/app/blob/a723af64acd5fd0415cffe6343e597b50e31dc2c/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.init.js#L222): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1067)
5. MAJOR [ve.init.wikia.ViewPageTarget.init.js#L266](https://github.com/Wikia/app/blob/a723af64acd5fd0415cffe6343e597b50e31dc2c/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.init.js#L266): Function has a complexity of 11 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
6. MAJOR [ve.init.wikia.ViewPageTarget.init.js#L267](https://github.com/Wikia/app/blob/a723af64acd5fd0415cffe6343e597b50e31dc2c/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.init.js#L267): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1067)
7. MAJOR [ve.init.wikia.ViewPageTarget.init.js#L276](https://github.com/Wikia/app/blob/a723af64acd5fd0415cffe6343e597b50e31dc2c/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.init.js#L276): Take the required action to fix the issue indicated by this comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1134)
8. MAJOR [ve.init.wikia.ViewPageTarget.init.js#L323](https://github.com/Wikia/app/blob/a723af64acd5fd0415cffe6343e597b50e31dc2c/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.init.js#L323): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1067)
9. MAJOR [ve.init.wikia.ViewPageTarget.init.js#L399](https://github.com/Wikia/app/blob/a723af64acd5fd0415cffe6343e597b50e31dc2c/extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.init.js#L399): Make this function anonymous by removing its name: 'function() {...}'. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
10. MAJOR [FlowTracking.hooks.php#L26](https://github.com/Wikia/app/blob/a723af64acd5fd0415cffe6343e597b50e31dc2c/extensions/wikia/FlowTracking/FlowTracking.hooks.php#L26): Remove the unused function parameters "$out, $skin". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)

**Review by cormorando — APPROVED** [review]

**Comment by mszabo-wikia** [comment]
LGTM

**Pull request merged**